helm/oauth2-proxy/templates/deployment.yaml

```diff
@@ -147,7 +147,13 @@ spec:
 {{ tpl (toYaml .Values.extraEnv) . | indent 8 }}
         {{- end }}
         ports:
+        {{- if (and (eq .Values.httpScheme "http") (empty .Values.containerPort)) }}
           - containerPort: 4180
+        {{- else if (and (eq .Values.httpScheme "https") (empty .Values.containerPort)) }}
+          - containerPort: 4443
```

Comment:
Its by default 443

<html>
<body>
<!--StartFragment-->

--https-address | string | <addr>:<port> to listen on for HTTPS clients | ":443"
-- | -- | -- | --


<!--EndFragment-->
</body>
</html>

https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview

Reply:
Unfortunately, I believe this would not work if securityContext is enabled (not running as root). Should we do something here, or accept the defaults?

Reply:
On another level, I'm afraid in the current form this will break existing installations (and thus is a breaking change). All releases using HTTPS will see the port on the Pod changing, but oauth2-proxy will still listen to the same old 4180 port.
I would suggest keeping it simple by just stating a port:
`- containerPort: {{ .Values.containerPort }}`
and a value, 4180 by default.

Reply:
@bluebrown can you please follow up on this?

Reply:
I think I mis-read the Service yaml. It is ok this way. I would just suggest changing default ports to 4443 and also override the defaults in an oauth2-proxy flag like it is done in https://github.com/oauth2-proxy/manifests/pull/117